### ISSUE TITLE
Build failure on macOS 10.13 (with Xcode 9)

### ISSUE THREAD

**Issue by fxcoudert** [actor-framework/actor-framework#600]
The error occurs when compiling `actor_config.cpp`:

```
[  0%] Building CXX object libcaf_core/CMakeFiles/libcaf_core_shared.dir/src/actor_config.cpp.o
In file included from /tmp/caf-20170929-82425-6x61di/actor-framework-0.15.3/libcaf_core/src/actor_config.cpp:20:
In file included from /tmp/caf-20170929-82425-6x61di/actor-framework-0.15.3/libcaf_core/caf/actor_config.hpp:23:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string:470:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/string_view:171:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__string:56:
In file included from /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/algorithm:637:
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:4291:23: error: 
      calling 'operator()' with incomplete return type 'caf::behavior'
_LIBCPP_INVOKE_RETURN(_VSTD::forward<_Fp>(__f)(_VSTD::forward<_Args>(__args)...))
~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__config:392:15: note: 
      expanded from macro '_VSTD'
#define _VSTD std::_LIBCPP_NAMESPACE
              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:4188:23: note: 
      expanded from macro '_LIBCPP_INVOKE_RETURN'
    noexcept(noexcept(__VA_ARGS__)) -> decltype(__VA_ARGS__) \
                      ^~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/type_traits:4308:9: note: 
      in instantiation of exception specification for '__invoke<std::__1::function<caf::behavior (caf::local_actor *)>
      &, caf::local_actor *>' requested here
        _VSTD::__invoke(_VSTD::declval<_Fp>(), _VSTD::declval<_Args>()...));
        ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/__config:392:15: note: 
      expanded from macro '_VSTD'
#define _VSTD std::_LIBCPP_NAMESPACE
              ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/functional:1591:33: note: 
      in instantiation of template class 'std::__1::__invokable_r<void, std::__1::function<caf::behavior
      (caf::local_actor *)> &, caf::local_actor *>' requested here
                                __invokable<_Fp&, _ArgTypes...>::value>
                                ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/functional:1616:9: note: 
      in instantiation of default argument for '__callable<std::__1::function<caf::behavior (caf::local_actor *)> >'
      required here
        __callable<_Fp>::value && !is_same<_Fp, function>::value
        ^~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/functional:1618:5: note: 
      in instantiation of default argument for 'function<std::__1::function<caf::behavior (caf::local_actor *)> >'
      required here
    function(_Fp);
    ^~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/functional:1578:28: note: 
      while substituting deduced template arguments into function template 'function' [with _Fp =
      std::__1::function<caf::behavior (caf::local_actor *)>, $1 = (no value)]
class _LIBCPP_TEMPLATE_VIS function<_Rp(_ArgTypes...)>
                           ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include/c++/v1/functional:1670:9: note: 
      'operator()' declared here
    _Rp operator()(_ArgTypes...) const;
        ^
/tmp/caf-20170929-82425-6x61di/actor-framework-0.15.3/libcaf_core/caf/fwd.hpp:62:7: note: forward declaration of
      'caf::behavior'
class behavior;
      ^
1 error generated.
make[3]: *** [libcaf_core/CMakeFiles/libcaf_core_shared.dir/src/actor_config.cpp.o] Error 1
make[2]: *** [libcaf_core/CMakeFiles/libcaf_core_shared.dir/all] Error 2
```

**Comment by sourcedelica** [comment]
I think this is a duplicate of #545, fixed in 0.15.4

**Comment by fxcoudert** [comment]
Thanks, will update in Homebrew!
